design/sketch.webidl

```diff
@@ -169,12 +169,12 @@ interface GPUTextureUsage {
 
 dictionary GPUTextureDescriptor {
     GPUExtent3D size;
-    u32 arrayLayerCount;
-    u32 mipLevelCount;
-    u32 sampleCount;
-    GPUTextureDimension dimension;
-    GPUTextureFormat format;
-    GPUTextureUsageFlags usage;
+    u32 arrayLayerCount = 1;
+    u32 mipLevelCount = 1;
```

Comment:
mipLevelCount should probably not have a default; the most sensible case (and the only option in Metal) is to default to the number of mips as calculated by width/height.

Reply:
Why would defaulting to some auto-computed value be most sensible? It's not like we provide auto-generation of mipmaps from level 0 anyway, so they need to find a way to provide the data or generate it, in which case clearly they need to know which levels they use.

Reply:
Metal has no way of specifying the number of mipmaps in a texture -- you can either choose "mipmapped" or "not mipmapped" [0]. I'm not sure how WebGPU is going to tackle this in general.

[0 ]https://developer.apple.com/documentation/metal/mtltexturedescriptor

Reply:
There's a read-write attribute mipmapLevelCount, too.

Reply:
yep, we absolutely do set the count on Metal: https://github.com/gfx-rs/gfx/blob/54f8da97c1ca02f4e2016a338dc450db12a10537/src/backend/metal/src/device.rs#L2212